MvvmCross-Plugins/PictureChooser/MvvmCross.Plugins.PictureChooser.Wpf/MvxPictureChooserTask.cs

```diff
@@ -65,6 +67,37 @@ public void ChoosePictureFromLibrary(int maxPixelDimension, int percentQuality,
             }
         }
 
+        /// <summary>
+        /// Get valid bitmap encoder from file extension.
+        /// </summary>
+        /// <param name="fileName">File name or file path, with extension</param>
+        /// <param name="percentQuality">quality level (for jpg files only)</param>
+        /// <returns></returns>
+        private BitmapEncoder GetBitmapEncoder(string fileName, int percentQuality)
+        {
+            if (fileName.ToLower().EndsWith("jpg") || fileName.ToLower().EndsWith("jpeg"))
```

Comment:
call `fileName.ToLower()` once and reuse, otherwise it will get called a lot of times in all of the checks.

You could also be a bit safer and use `Path` to grab the extension.